forge-game/src/main/java/forge/game/ability/effects/EffectEffect.java

```diff
@@ -322,6 +322,12 @@ public void run() {
                     game.getUpkeep().addUntil(controller, endEffect);
                 } else if (duration.equals("UntilEndOfCombat")) {
                     game.getEndOfCombat().addUntil(endEffect);
+                } else if (duration.equals("UntilEndOfYourNextCombat")) {
+                    if (game.getPhaseHandler().inCombat()) {
```

Comment:
@tool4ever or anyone else – can anyone clue me towards what's wrong here?

Need to 🛌 but will debug more in Phase.java later

Reply:
Currently `registerUntilEnd` is not supported automatically for all PhaseType

But I think this wouldn't be good enough anyway, if you check the duration it has a slightly weirder wording:
it doesn't say "next combat" but "combat on your next turn"
so this would break when an additional combat follows after

It probably needs to be chained like with "ThisTurnAndNextTurn" (untested):
`getEndOfTurn().registerUntilEnd(controller) -> getEndOfCombat().addUntilEnd()`